metis/embedders/jina_v4.py

```
@@ -47,26 +47,24 @@ def __init__(self, config: Optional[EmbeddingConfig] = None) -> None:
-            logger.warning("Could not determine embedding dimension, using default 2048")
-            self._dim = 2048
+        # Force Jina v4 to use 2048 dimensions (full fidelity)
+        # The model supports 512/768/2048, but we want max dimensional fidelity
+        self._dim = 2048
+        logger.info(f"Jina v4 embedder initialized with {self._dim} dimensions")
+
         # Jina v4 tolerates long contexts; allow wider chunks by default
         self.chunk_size_tokens = self.config.chunk_size_tokens or 1024
         self.chunk_overlap_tokens = self.config.chunk_overlap_tokens or 256
 
+    def _infer_embedding_dim(self, default: int) -> int:
```

Ruff (0.13.3)

Unused method argument: `default`

(ARG002)